include/extractor/guidance/road_classification.hpp

```diff
@@ -0,0 +1,121 @@
+// a road simply offered for connectivity. Will be ignored in forks/other decisions. Always
+// considered non-obvious to continue on
+const constexpr Enum CONNECTIVITY = 31;
+} // namespace Road Class
+
+class RoadClassification
+{
+    // a class that behaves like a motorway (separated directions)
+    std::uint32_t motorway_class : 1;
+    // all types of link classes
+    std::uint32_t link_class : 1;
+    // a low priority class is a pure connectivity way. It can be ignored in multiple decisions
+    // (e.g. fork on a primary vs service will not happen)
+    std::uint32_t may_be_ignored : 1;
+    // the road priority is used as an indicator for forks. If the roads are of similar priority
+    // (difference <=1), we can see the road as a fork. Else one of the road classes is seen as
+    // obvious choice
+    RoadPriorityClass::Enum road_priority_class : 5;
+
+  public:
+    // default construction
+    RoadClassification()
+        : motorway_class(0), link_class(0), may_be_ignored(1),
+          road_priority_class(RoadPriorityClass::CONNECTIVITY)
+    {
+    }
+
+    RoadClassification(bool motorway_class,
+                       bool link_class,
+                       bool may_be_ignored,
+                       RoadPriorityClass::Enum road_priority_class)
+        : motorway_class(motorway_class), link_class(link_class), may_be_ignored(may_be_ignored),
+          road_priority_class(road_priority_class)
+    {
+    }
+
+    inline bool IsMotorwayClass() const { return (0 != motorway_class) && (0 == link_class); }
+    inline void SetMotorwayFlag(const bool new_value) { motorway_class = new_value; }
+
+    inline bool IsRampClass() const { return (0 != motorway_class) && (0 != link_class); }
+
+    inline bool IsLinkClass() const { return (0 != link_class); }
+    inline void SetLinkClass(const bool new_value) { link_class = new_value; }
+
+    inline bool IsLowPriorityRoadClass() const { return (0 != may_be_ignored); }
+    inline void SetLowPriorityFlag(const bool new_value) { may_be_ignored = new_value; }
+
+    inline std::uint32_t GetPriority() const
+    {
+        return static_cast<std::uint32_t>(road_priority_class);
+    }
+
+    inline RoadPriorityClass::Enum GetClass() const { return road_priority_class; }
+    inline void SetClass(const RoadPriorityClass::Enum new_value)
+    {
+        road_priority_class = new_value;
+    }
+
+    inline bool operator==(const RoadClassification &other) const
+    {
+        return motorway_class == other.motorway_class && link_class == other.link_class &&
+               may_be_ignored == other.may_be_ignored &&
+               road_priority_class == other.road_priority_class;
+    }
+
+    inline std::string ToString() const
+    {
+        return std::string() + (motorway_class ? "motorway" : "normal") +
+               (link_class ? "_link" : "") + (may_be_ignored ? " ignorable " : " important ") +
+               std::to_string(road_priority_class);
+    }
```

Comment:
as discussed in chat a couple of days ago: member functions defined inside a class are inline by default